vllm/config.py

```diff
@@ -1423,10 +1424,15 @@ def __post_init__(self) -> None:
         self.world_size = self.pipeline_parallel_size * \
             self.tensor_parallel_size
 
-        self.data_parallel_size = envs.VLLM_DP_SIZE
-        self.data_parallel_rank = envs.VLLM_DP_RANK
-        self.data_parallel_master_ip = envs.VLLM_DP_MASTER_IP
-        self.data_parallel_master_port = envs.VLLM_DP_MASTER_PORT
+        if self.data_parallel_size > 1:
+            self.data_parallel_master_port = get_open_port()
```

Comment:
will this function be executed again in every process, and then they have different `data_parallel_master_port` ?

Reply:
@youkaichao I don't think so, since this is in `__post_init__` and the config objects are passed to the subprocesses already constructed.